src/vmm/src/devices/virtio/net/device.rs

```diff
@@ -553,9 +556,11 @@ impl Net {
             }
         }
 
-        // At this point we processed as many Rx frames as possible.
-        // We have to wake the guest if at least one descriptor chain has been used.
-        self.signal_used_queue(NetQueue::Rx)
+        // We only notify guest if there is at least one packet processed successfully.
+        if signal_queue {
+            self.signal_used_queue(NetQueue::Rx)?;
```

Comment:
We still only signal once, no matter how many descriptors we put into the used ring, right? So we're still not completely spec compliant:

> After the device writes a descriptor index into the used ring:
> – If flags is 1, the device SHOULD NOT send a notification.
> – If flags is 0, the device MUST send a notification

(from Section 2.6.7.2: https://docs.oasis-open.org/virtio/virtio/v1.1/csprd01/virtio-v1.1-csprd01.pdf)

Reply:
This is from `If the VIRTIO_F_EVENT_IDX feature bit is not negotiated: ` section. But for us, the `VIRTIO_F_EVENT_IDX` is basically always enabled.
According to this:
```
After the device writes a descriptor index into the used ring:

    If the idx field in the used ring (which determined where that descriptor index was placed) was equal to used_event, the device MUST send a notification.
    Otherwise the device SHOULD NOT send a notification.
```
I think we are quite complaint.